controller/internal/controller/consumer_controller.go

```diff
@@ -181,3 +197,73 @@ func (r *ConsumerReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		)
 	return controller.Complete(r)
 }
+
+// checkConsumerConflicts Perform full conflict detection and rebuild the index
+func (r *ConsumerReconciler) checkConsumerConflicts(ctx context.Context) error {
```

Comment:
We should check conflicts when reconciling the CR: https://github.com/mosn/htnn/blob/4568dd3888f17d275b2a73c6ee66683ec19796f7/controller/internal/controller/consumer_controller.go#L54, instead of manually calling the  List method.